File: src/buildjs/Model/Team.cs
namespace buildjs.Model
{
    public class Team
    {
        public Team()
        {
            IsVisible = true;
        }

        public int Id { get; set; }
        public string Name { get; set; }

        public string SubTitle { get; set; }

        public int ForegroundColour { get; set; }
        public int BackgroundColour { get; set; }
        public bool IsVisible { get; set; }
    }
}

File: src/buildjs/src/team.ts
import {inject} from 'aurelia-framework';
import {HttpClient} from 'aurelia-http-client';
import {TeamViewModel} from './models/team-view-model';

@inject(HttpClient)
export class Team {

    httpClient: HttpClient;
    team: TeamViewModel;
    
  constructor(httpClient: HttpClient) {
    this.httpClient = httpClient;
    this.team = new TeamViewModel({"ForegroundColour": 1, "BackgroundColour": 1})       
  }
  
  activate(params: any) {
    var teamId = params.id;
    
    return this.httpClient.get('./team/'+teamId)
      .then(response => {
        this.team = new TeamViewModel(response.content);
      });
    
  }
}
